crates/e2e/tests/e2e/app_data.rs

```diff
@@ -105,5 +105,60 @@ async fn app_data(web3: Web3) {
     let order3_ = services.get_order(&uid).await.unwrap();
     assert_eq!(order3_.data.app_data, app_data_hash);
     // Contrast this with order0, which doesn't have full app data.
-    assert_eq!(order3_.metadata.full_app_data, Some(app_data.to_string()));
+    assert_eq!(order3_.metadata.full_app_data.as_deref(), Some(app_data));
+
+    // invalid app data
+    let invalid_app_data = r#"{"backend":"invalid"}"#;
+    let order4 = create_order(OrderCreationAppData::Full {
+        full: invalid_app_data.to_string(),
+    });
+    let err = services.create_order(&order4).await.unwrap_err();
+    dbg!(err);
```

Comment:
left over `dbg!()` statement? (2 more in this test)

Reply:
I kept these in on purpose, as the rest of the test uses this pattern (and assumed it was for a reason - specifically to manually check the error contents when running the test with `--nocapture`).